metrics/consensuslatency.go

```diff
@@ -0,0 +1,70 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/relab/hotstuff"
+	"github.com/relab/hotstuff/eventloop"
+	"github.com/relab/hotstuff/logging"
+	"github.com/relab/hotstuff/metrics/types"
+	"github.com/relab/hotstuff/modules"
+)
+
+func init() {
+	RegisterReplicaMetric("consensus-latency", func() any {
+		return &ConsensusLatency{}
+	})
+}
+
+// ConsensusLatency processes ConsensusLatencyMeasurementEvent, and
+// writes ConsensusLatencyMeasurementEvent to the metrics logger.
+type ConsensusLatency struct {
+	metricsLogger Logger
+	opts          *modules.Options
+
+	wf Welford
+}
+
+// InitModule gives the module access to the other modules.
+func (lr *ConsensusLatency) InitModule(mods *modules.Core) {
+	var (
+		eventLoop *eventloop.EventLoop
+		logger    logging.Logger
+	)
+
+	mods.Get(
+		&lr.metricsLogger,
+		&lr.opts,
```

Comment:
There is only one use of `lr.opts` on Line 63: `uint32(lr.opts.ID())`. We could instead make `id` be part of the struct and initialize it in `InitModule` from a local field `opts`. This would save us from having `opts` be part of the struct.